specs/latest/2.0/index.html

```diff
@@ -1370,40 +1479,31 @@ <h2 class="no-toc">Table of contents</h2>
         <p>If a WebGLBuffer is bound to the <code>PIXEL_UNPACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code> error.</p>
         <p>Sized internal formats are supported in WebGL 2 and <em>internalformat</em> is no longer required to be the same as <em>format</em>. Instead, the combination of <em>internalformat</em>, <em>format</em>, and <em>type</em> must be listed in Table 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
-        <p>If <em>type</em> is specified as <code>FLOAT_32_UNSIGNED_INT_24_8_REV</code>, <em>pixels</em> must be null; otherwise, generates an <code>INVALID_OPERATION</code> error.</p>
-        <p>If <em>pixels</em> is non-null, the type of <em>pixels</em> must
-match the <em>type</em> according to the following table; otherwise, generates an <code>INVALID_OPERATION</code> error:</p>
-        <table id="TEXTURE_PIXELS_TYPE_TABLE">
-          <tr><th>type of <em>pixels</em></th><th><em>type</em></th></tr>
-          <tr><td>Int8Array</td><td>BYTE</td></tr>
-          <tr><td>Uint8Array</td><td>UNSIGNED_BYTE</td></tr>
-          <tr><td>Int16Array</td><td>SHORT</td></tr>
-          <tr><td>Uint16Array</td><td>UNSIGNED_SHORT</td></tr>
-          <tr><td>Uint16Array</td><td>UNSIGNED_SHORT_5_6_5</td></tr>
-          <tr><td>Uint16Array</td><td>UNSIGNED_SHORT_5_5_5_1</td></tr>
-          <tr><td>Uint16Array</td><td>UNSIGNED_SHORT_4_4_4_4</td></tr>
-          <tr><td>Int32Array</td><td>INT</td></tr>
-          <tr><td>Uint32Array</td><td>UNSIGNED_INT</td></tr>
-          <tr><td>Uint32Array</td><td>UNSIGNED_INT_5_9_9_9_REV</td></tr>
-          <tr><td>Uint32Array</td><td>UNSIGNED_INT_2_10_10_10_REV</td></tr>
-          <tr><td>Uint32Array</td><td>UNSIGNED_INT_10F_11F_11F_REV</td></tr>
-          <tr><td>Uint32Array</td><td>UNSIGNED_INT_24_8</td></tr>
-          <tr><td>Uint16Array</td><td>HALF_FLOAT</td></tr>
-          <tr><td>Float32Array</td><td>FLOAT</td></tr>
-        </table>
         <p>
             If <a href="#PIXEL_STORE_PARAM_CONSTRAINTS">pixel store parameter constraints</a> are not met,
             generates an <code>INVALID_OPERATION</code> error.
         </p>
+        <p>If there's not enough data in <code>srcData</code> starting at <code>srcOffset</code>,
+           generate INVALID_OPERATION.
       </dd>
 
-      <dt class="idl-code">void texImage2D(GLenum target, GLint level, GLint internalformat, GLenum format, GLenum type, TexImageSource? source) /* May throw DOMException */
+      <dt class="idl-code">
+        [throws]
+        void texImage2D(GLenum target, GLint level, GLint internalformat,
+                        GLsizei width, GLsizei height, GLint border,
+                        GLenum format, GLenum type,
+                        TexImageSource source) // May throw DOMException
```

Comment:
Cleaving closer to the GL spec keeps things simpler. I think for a large API like this, that's more important.
Wrappers to change this functionality are easy to write.